tun/tun_linux.go

```diff
@@ -399,6 +491,77 @@ func (tun *NativeTun) Close() error {
 	return err2
 }
 
+func (tun *NativeTun) BatchSize() int {
+	return tun.batchSize
+}
+
+func (tun *NativeTun) DisableOffload() error {
```

Comment:
Not needed, if we're getting rid of that knob.

Reply:
We would like to keep this for broader coverage of customer concerns. There exist network drivers that are buggy, and we have users running on the class of hardware where this more often comes up (such as embedded devices with small oem vendor drivers attempting to optimize for their nics). If an end user uses us on such a device, it may be rare for them to be able to get their device OEM to talk to the soc OEM to get them to provide a new patch for the driver. I've personally lived through this with unifi's Cavium devices having bad nic drivers. I fully understand the desire for wireguard-go standalone not wanting to be encumbered with this smaller group of users and the exposing of a tuning knob, but our interesting in that smaller group of customers may be different. If we never discover such a user in practice I'm happy to come back and remove this API, but the cost of ownership for wireguard-go here should be very low, especially with the tuning knob removed.

Reply:
But this is all tun side, not physical packet side, right? It [un]sets `NETIF_F_TSO` and `NETIF_F_TSO6` and such on the virtual tun driver, not on some hardware driver.

Reply:
Removed in https://github.com/WireGuard/wireguard-go/pull/64/commits/1ac54b1e36950a9b4369d3575e1117884e3521b7. We have yet to encounter any offloading bugs. This API, while convenient, is not necessary to disable offloading if the need arises.

Reply:
> But this is all tun side, not physical packet side, right? It [un]sets NETIF_F_TSO and NETIF_F_TSO6 and such on the virtual tun driver, not on some hardware driver.

Yes, though in the case of forwarding the offload can extend to/from a physical netdev.